### PR TITLE
test: split settings renderers

### PR DESCRIPTION
**Pull request by CyanAutomation**
## Summary
- split settings renderer tests into dedicated files
- add per-renderer edge cases and muted console wrappers

## Testing
- `npx prettier . --check`
- `npx eslint .`
- `npm run check:jsdoc`
- `npx vitest run`
- `npx playwright test` *(fails: Next readiness only in cooldown, Next button cooldown skip)*
- `npm run check:contrast`

------
https://chatgpt.com/codex/tasks/task_e_68bc73848e3483268e9b51e33270486e

### CODE REVIEW

File: tests/helpers/syncDisplayMode.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    }));
+    vi.mock("../../src/helpers/viewTransition.js", () => ({
+      withViewTransition: vi.fn((fn) => fn())
+    }));
+    const { syncDisplayMode } = await import("../../src/helpers/settings/syncDisplayMode.js");
+    document.getElementById("display-mode-dark").checked = true;
+    const handleUpdate = vi.fn().mockResolvedValue();
+    const updated = await withMutedConsole(() =>
+      syncDisplayMode({ displayMode: "light" }, handleUpdate)
+    );
```
[the remark this answers is not in the document]
Comment (CyanAutomation) [reply tests/helpers/syncDisplayMode.test.js:27]: @codex fix comments
